src/components/Editor/HighlightLine.js

```diff
@@ -43,14 +44,50 @@ function isDocumentReady(selectedSource, selectedLocation) {
 }
 
 export class HighlightLine extends PureComponent<Props> {
+  isInSteppingUpdate: boolean;
+  previousEditorLine: ?number;
+
+  constructor(props: Props) {
+    super(props);
+    this.isInSteppingUpdate = false;
+    this.previousEditorLine = null;
+  }
+
+  shouldComponentUpdate(nextProps: Props) {
+    const { selectedLocation, selectedSource } = nextProps;
+    return this.shouldSetHighlightLine(selectedLocation, selectedSource);
+  }
+
+  shouldSetHighlightLine(
+    selectedLocation: Location,
+    selectedSource: SourceRecord
+  ) {
+    const { sourceId, line } = selectedLocation;
+    const editorLine = toEditorLine(sourceId, line);
+    if (this.isInSteppingUpdate && editorLine === this.previousEditorLine) {
+      return false;
+    }
+    if (!isDocumentReady(selectedSource, selectedLocation)) {
+      return false;
+    }
+    return true;
+  }
+
   componentDidUpdate(prevProps: Props) {
-    const { selectedLocation, selectedFrame, selectedSource } = this.props;
+    const {
+      pauseCommand,
+      selectedLocation,
+      selectedFrame,
+      selectedSource
+    } = this.props;
+    if (["resume", "stepIn", "stepOver", "stepOut"].includes(pauseCommand)) {
```

Comment:
i think this is all of the options :)
could we do something like `!!pauseCommand` which will be the same?